kbase-extension/static/kbase/js/widgets/function_output/kbaseRESKESearchResultDemo.js

```diff
@@ -0,0 +1,535 @@
+                        </row>
+
+                    </div>
+                </div>
+            `);   
+
+
+            self.$container.append(self.$searchObjectsResultPane);
+            var $searchResultPanel = $("#searchResultPanel");
+
+            $.each( searchObjectsOutput.objects, function(i, value){
+                $searchResultPanel.append($(
+                    `
+                      <div class="row" style="margin-top:30px;">
+                        <div class="col col-sm-1">
+                        </div>
+                        <div class="col col-sm-1">
+                            <b>${ (i+start) }. </b>
+                        </div>
+                        <div id="row${i}" class="col col-sm-8" style="background-color:#EEFFEE;">
+                            <b>GUID:</b> ${value.guid} <br>
+                            <b>object_name:</b> ${value.object_name} <br>
+                        </div>
+                        <div class="col col-sm-3">
+                        </div>
+                      </div>
+                    `
+                ));
+                var $row = $(`#row${i}`);
+                Object.keys(value.key_props).forEach(function (key) {
+                    var val = self.escapeHtml(value.key_props[key]);
+                    if(val.length > 100){
+                        val = val.substring(0,100) + "...";
+                    }
+                    $row.append($(` <b>${key}:</b> ${val}  <br> `));
+                 });
+
+
+            });
+            $searchResultPanel.append($('<hr>'));
+            self.buildPagenator($searchResultPanel, searchObjectsOutput);
+
+
+            var $searchTypeRestulsListGroup = $("#searchTypeRestulsListGroup");
+        },
+
+
+        loggedInCallback: function(event, auth) {
+
+            this.genericClient = new GenericClient(this.searchEngineURL, auth, null, false);
+            this.render();
+            return this;
+        },
+
+        render: function(){
+            var self = this;
+            var $container = $("<div/>");
+            self.$elem.append( $container );
+            self.$container = $container;
+
+            self.loading(false);
+            self.buildKBaseSearchPanel();
+        },
+
+        loading: function(isLoading) {
+            if (isLoading)
+                this.showMessage("<img src='" + loadingImage + "'/>");
+            else
+                this.hideMessage();
+        },
+
+        showMessage: function(message) {
+            var span = $("<span/>").append(message);
+
+            this.$messagePane.append(span);
+            this.$messagePane.show();
+        },
+
+        hideMessage: function() {
+            this.$messagePane.hide();
+            this.$messagePane.empty();
+        },
+
+        clientError: function(error){
+            this.loading(false);
+            this.showMessage(error.error.error);
+        },        
+
+        escapeHtml:  function(string) {
```

Comment:
This seems useful enough to move to an API call, probably in util/string.js. But we can do that later.